lib/iex/endpoints/historial_prices.rb

```diff
@@ -0,0 +1,21 @@
+module IEX
+  module Endpoints
+    module HistoricalPrices
+      def historical_prices(symbol, options = {})
+        slug = "stock/#{symbol}/chart"
+        slug += "/#{options[:range]}" if options[:range]
+        slug += "/#{options[:date]}" if options[:range] == 'date'
+
+        # We only want options to include query params at this point, remove :range and :date
+        options.delete(:range)
+        options.delete(:date)
```

Comment:
You're modifying the hash passed in, so either `options = options.dup`, or use `.except` or similar to get a copy without those keys. Let's make sure there are no other places where we did this, I may have missed that.

Reply:
I didn't see any downside to modifying it but happy to dup it if you prefer. This was the only place I did this.

Reply:
The downside is that code like this doesn't do what you'd expect it to do:

```ruby
options = { range: ... }
prices_for_msft = historical_prices('msft', options)
prices_for_goog = historical_prices('goog', options)
```

Reply:
Also use `options.key?(:range)`, `nil` is a value and while it doesn't make sense you don't want the caller to rely on you clearing the option, this is more by convention

Reply:
Yup that's a downside. Thanks